### ISSUE TITLE
kube-derive types don't generate kind/apiVersion

### ISSUE THREAD

**Issue by ctron** [clux/kube-rs#315]
A type defined by `k8s-openapi` provides `kind`/`apiVersion`:
~~~yaml
apiVersion: ""
kind: ""
metadata: {}
spec: {}
~~~

A type defined with `kube-derive` does not provide `kind`/`apiVersion`
~~~yaml
apiVersion: v1
kind: ServiceAccount
metadata: {}
~~~

Reproducer:

~~~rust
println!(
    "{}",
    serde_yaml::to_string(&ServiceAccount {
        ..Default::default()
    })?
);

println!(
    "{}",
    serde_yaml::to_string(&Foo {
        ..Default::default()
    })?
);
~~~

My expectation would be, that `kube-derive` types do provide the same information.

**Comment by clux** [comment]
This was only a problem when using the `#[kube(derive = "Default")]` and using `Foo::default` rather than `Foo::new` (which is a reasonable thing to do. We now manually specialise default for the generated type if the above instruction is given.

**Comment by clux** [comment]
Released in 0.42.0. Thanks for the issue!

**Comment by ctron** [comment]
> Released in 0.42.0. Thanks for the issue!

Looks good, thanks for the fix!
